packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp

```diff
@@ -447,6 +466,58 @@ namespace MueLu {
     prec_->compute();
   }
 
+  template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
+  void Ifpack2Smoother<Scalar, LocalOrdinal, GlobalOrdinal, Node>::SetupBlockRelaxation(Level& currentLevel) {
+    typedef Tpetra::RowMatrix<SC,LO,GO,NO> tRowMatrix;
+
+    RCP<BlockedCrsMatrix> bA = rcp_dynamic_cast<BlockedCrsMatrix>(A_);
+    if (!bA.is_null())
+      A_ = bA->Merge();
+
+    RCP<const tRowMatrix> tA = Utilities::Op2NonConstTpetraRow(A_);
+
+    bool reusePreconditioner = false;
+    if (this->IsSetup() == true) {
+      // Reuse the constructed preconditioner
+      this->GetOStream(Runtime1) << "MueLu::Ifpack2Smoother::SetupGeneric(): Setup() has already been called, assuming reuse" << std::endl;
+
+      RCP<Ifpack2::Details::CanChangeMatrix<tRowMatrix> > prec = rcp_dynamic_cast<Ifpack2::Details::CanChangeMatrix<tRowMatrix> >(prec_);
+      if (!prec.is_null()) {
+#ifdef IFPACK2_HAS_PROPER_REUSE
```

Comment:
I'm not sure what the interaction of reuse with `Merge()` call before. Somebody needs to convince me that it would do the right thing here when called in reuse mode. Plus, the reuse in line smoothing also needs checking.

Reply:
This question is actually outside of the scope of my work, I am merely providing the same functionality as the one provided in `SetupGeneric()`.
I am not against removing all the reuse here but that also means that all BLOCK_RELAXATION smoothers will have no reuse.
Also if the logic is flawed then `SetupGeneric()` needs to be modified...

Reply:
That's the valid point. Git blame shows that it was me who added the `Merge()` in `SetupGeneric`. So, you are right, it may be disregarded. The question of whether it actually works with reuse is orthogonal to your work.